src/linear_algebra/matrix/charpoly/eigs.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Mohanad Ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import data.polynomial.basic
+import field_theory.is_alg_closed.basic
+import algebra.char_p.two
+
+/-!
+# Eigenvalues are characteristic polynomial roots.
+
+## Two Lemmas
+In algebraically closed fields the following two lemmas are available:
+  (or when one is willing to extend the working field to such fields like ℝ to ℂ):
+- `det_eq_prod_roots_charpoly` determinant is the product of the roots of the characteristic
+polynomial.
+- `trace_eq_sum_roots_charpoly` trace of a matrix is the sum of the roots of the characteristic
+polynomial.
+-/
+variables {n: Type*}[fintype n][decidable_eq n]
+variables {R: Type*}[field R][is_alg_closed R]
+
+open matrix polynomial
+open linear_map module.End
+open_locale matrix big_operators
+
+lemma det_eq_prod_roots_charpoly (A : matrix n n R):
```

Comment:
This name is backwards,
```suggestion
lemma prod_roots_charpoly_eq_det (A : matrix n n R) :
```

Reply:
Usually the convention is to put the simpler expression on the RHS, which is `det` or `trace`. When constructing a query searching mathlib it's expected that you apply this heuristic too.

Reply:
You mean simpler in length? Or conceptually simpler?

How about these theorems in mathlib:
- [matrix.trace_eq_neg_charpoly_coeff](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/charpoly/coeff.html#matrix.trace_eq_neg_charpoly_coeff)
- [matrix.det_eq_sign_charpoly_coeff](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/charpoly/coeff.html#matrix.det_eq_sign_charpoly_coeff)
- [matrix.is_hermitian.det_eq_prod_eigenvalues](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/spectrum.html#matrix.is_hermitian.det_eq_prod_eigenvalues)
- [matrix.is_hermitian.eigenvalues_eq](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/spectrum.html#matrix.is_hermitian.eigenvalues_eq)

I think the simpler side is the LHS in the lemmas mentioned above.

In the derivations I am imagining the user will probably consider the eigenvalues (charpoly roots) more simple as they allow him to do further manipulations. In that sense "determinant equals product of eigenvalues" is more like a "definition" you "unfold" to do more work with, than a simp theorem that gives a simplification on its right hand side. For example consider proving `det(A) < det(1 + A)` for a positive semidefinite matrix because every eigenvalue is increased by 1 on the RHS.

Anyways, I will defer to your experience on this one and reverse the lemma names.

Reply:
Ah, those first two lemmas are a good counterargument. Feel free to use that order after all.

Just make sure the lemma name and statement match!